crates/proof-of-sql-planner/src/util.rs

```diff
@@ -0,0 +1,243 @@
+use super::{PlannerError, PlannerResult};
+use arrow::datatypes::{Field, Schema};
+use datafusion::{
+    catalog::TableReference,
+    common::{Column, DFSchema, ScalarValue},
+};
+use proof_of_sql::base::database::{ColumnField, ColumnRef, ColumnType, LiteralValue, TableRef};
+
+/// Convert a [`TableReference`] to a [`TableRef`]
+///
+/// If catalog is provided it is ignored
```

Comment:
What about this query: `(SELECT * FROM foo.bar.baz) UNION (SELECT * FROM qux.bar.baz)`?  That's what catelog means, right?
If a user provides a catalog, I'd rather give them an error than silently drop it, so I feel like this should return a `Result` and the `TableReference::Full` should error.